From b1e3360f90b7e2832b83c3b1f9b7640f79ac30cd Mon Sep 17 00:00:00 2001
From: Alexander Georgievskiy <galeksandrp@gmail.com>
Date: Sun, 20 Jun 2021 10:47:30 +0300
Subject: [PATCH] Add TX power in dBm options in web interface

remove comment from html templates

In some cases, this broke the markup. For example when you enable Smart Connect option.
---
 lib/wifi/qcawificfg80211.sh                  | 12 ++-
 usr/lib/lua/luci/view/web/apsetting/wifi.htm | 60 +++++++++++++
 usr/lib/lua/luci/view/web/inc/wifi.html      | 60 +++++++++++++
 usr/lib/lua/luci/view/web/setting/wifi.htm   | 90 ++++++++++++++++++++
 4 files changed, 221 insertions(+), 1 deletion(-)

diff --git a/lib/wifi/qcawificfg80211.sh b/lib/wifi/qcawificfg80211.sh
index 9a4ecf4..b65ee92 100644
--- a/etc/config/xqled
+++ b/etc/config/xqled
@@ -38,12 +38,12 @@
 #             MAX -> MIN : 7 - 1, temp reserve
 config func 'sys_booting'
     option nled     'sys'
-    option color    'yellow'
+    option color    'blue'
     option trigger  'on'
     option priority '7'

 config func 'sys_ok'
-    option nled     'sys'
+    option nled     'func'
     option color    'blue'
     option trigger  'on'
     option priority '1'
--- a/lib/wifi/qcawificfg80211.sh
+++ b/lib/wifi/qcawificfg80211.sh
@@ -2932,13 +2932,16 @@
 		# the latter doesn't exist
 
 		# for miwifi
 		if [ "$bdmode" = "24G" ]; then
-			max_power=30
+			max_power=20
 			wifitool "$ifname" setUnitTestCmd 67 3 16 1 1
 			iwpriv "$ifname" 11ngvhtintop 1
 		else
-			max_power=30
+			if [ "$channel" -ge 100 ]; then
+				max_power=26
+			else
+				max_power=23
 		fi
 
 		config_get txpwr "$device" txpwr
 		if [ "$txpwr" = "mid" ]; then
@@ -3750,12 +3753,12 @@
 	option channel	auto
 	option macaddr	$(cat /sys/class/net/${dev}/address)
 	option hwmode	11${mode_11}
 	option htmode	'${htmode}'
-	option country	'$country_code'
+	option country	'FR'
 	option disabled '$disable'
 	option txbf '3'
-	option ax '1'
+	option ax '0'
 EOF
 	if [ $devidx = 2 ]; then
 		cat <<EOF
 	option bw 20
@@ -3774,9 +3777,9 @@
 
 EOF
 	if [ $devidx = 0 ]; then
 		cat <<EOF
-	option channel_block_list '36,40,44,48,52,56,60,64'
+	option channel_block_list '64'
 EOF
 	fi
 	devidx=$(($devidx + 1))
 	done
